### ISSUE TITLE
error: failed to run custom build command for `ring v0.17.2`

### ISSUE THREAD

**Issue by carlosaherrera** [briansmith/ring#1720]
I'm trying to compile on MacOS 11.7.10 with Intel Core i7. Not sure how to get past this error.

```
$ cargo build

Compiling ring v0.17.2
error: failed to run custom build command for `ring v0.17.2`

Caused by:
  process didn't exit successfully: `/Users/foo/bar/baz/target/debug/build/ring-e81c8e069ee a9131/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=RING_PREGENERATE_ASM
  cargo:rustc-env=RING_CORE_PREFIX=ring_core_0_17_2_
  OPT_LEVEL = Some("0")
  TARGET = Some("x86_64-apple-darwin")
  HOST = Some("x86_64-apple-darwin")
  cargo:rerun-if-env-changed=CC_x86_64-apple-darwin
  CC_x86_64-apple-darwin = None
  cargo:rerun-if-env-changed=CC_x86_64_apple_darwin
  CC_x86_64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  CARGO_CFG_TARGET_FEATURE = Some("cmpxchg16b,fxsr,sse,sse2,sse3,ssse3")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-apple-darwin
  CFLAGS_x86_64-apple-darwin = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_apple_darwin
  CFLAGS_x86_64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None

  --- stderr
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/chacha-x86_64-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/chacha-x86_64-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/aesni-x86_64-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/aesni-x86_64-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/vpaes-x86_64-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/vpaes-x86_64-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/x86_64-mont-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/x86_64-mont-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/x86_64-mont5-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/x86_64-mont5-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/p256-x86_64-asm-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/p256-x86_64-asm-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/aesni-gcm-x86_64-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/aesni-gcm-x86_64-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/ghash-x86_64-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/ghash-x86_64-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/sha512-x86_64-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/sha512-x86_64-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/chacha20_poly1305_x86_64-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/chacha20_poly1305_x86_64-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/sha256-x86_64-macosx.o" "/Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/pregenerated/sha256-x86_64-macosx.S"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/curve25519.o" "crypto/curve25519/curve25519.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/aes_nohw.o" "crypto/fipsmodule/aes/aes_nohw.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/montgomery.o" "crypto/fipsmodule/bn/montgomery.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/montgomery_inv.o" "crypto/fipsmodule/bn/montgomery_inv.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/ecp_nistz.o" "crypto/fipsmodule/ec/ecp_nistz.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/gfp_p256.o" "crypto/fipsmodule/ec/gfp_p256.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/gfp_p384.o" "crypto/fipsmodule/ec/gfp_p384.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/p256.o" "crypto/fipsmodule/ec/p256.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/limbs.o" "crypto/limbs/limbs.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/mem.o" "crypto/mem.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/poly1305.o" "crypto/poly1305/poly1305.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/crypto.o" "crypto/crypto.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/cpu_intel.o" "crypto/cpu_intel.c"
  running "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "include" "-I" "/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out" "-Wall" "-Wextra" "-std=c1x" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wbad-function-cast" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wnested-externs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wstrict-prototypes" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-gfull" "-DNDEBUG" "-c" "-o/Users/foodev/baz/target/debug/build/ring-8d888a09f68cd256/out/curve25519_64_adx.o" "crypto/curve25519/curve25519_64_adx.c"
  crypto/curve25519/curve25519_64_adx.c:21:32: error: unknown warning group '-Wunused-but-set-parameter', ignored [-Werror,-Wunknown-warning-option]
  #pragma GCC diagnostic ignored "-Wunused-but-set-parameter"
                                 ^
  1 error generated.
  thread 'main' panicked at /Users/foo.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.17.2/build.rs:707:9:
  execution failed
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by briansmith** [comment]
My guess is that the best thing to do would be to upgrade Xcode and/or macOS if you can. Regardless, I will release 0.17.3 to address this. I made a PR #1721 towards that end.

**Comment by briansmith** [comment]
I published 0.17.3. Please try it out and let me know if it works better for you. (This issue got closed automatically because it was linked from the PR, but I can reopen it.)

**Comment by carlosaherrera** [comment]
Building with 0.17.3 solved my issue :). Thanks for the quick turn around.

> My guess is that the best thing to do would be to upgrade Xcode and/or macOS if you can

I would, but unfortunately MacOS 11.7.10 is the latest version supported by my MacBook Pro (Retina, 15-inch, Mid 2014).
